slsim/Sources/galaxies.py

```diff
@@ -55,8 +77,12 @@ def draw_source(self):
 
         if galaxy["e1"] == -1 or galaxy["e2"] == -1:
             e1, e2 = galaxy_projected_eccentricity(float(galaxy["ellipticity"]))
-            galaxy["e1"] = e1
-            galaxy["e2"] = e2
+            if self.list_type == "astropy_table":
+                galaxy["e1"] = e1
+                galaxy["e2"] = e2
+            else:
+                galaxy["e1"] = [e1]
```

Comment:
it is not clear to me that there are two formats required. What is the reason for it?

Reply:
Solved this issue!